src/api.rs

```diff
@@ -1000,14 +1397,42 @@ pub async fn gg18_sign_client_round6(context: String, delay: u32) -> String {
     let (phase5_com2, phase_5d_decom2) = context
         .local_sig
         .clone()
-        .unwrap()
+        .ok_or(TssError::UnknownError {
+            msg: ("gg18_sign_client_round6").to_string(),
+            file: file!().to_string(),
+            line: line!(),
+        })?
         .phase5c(
             &phase_5a_decomm_vec,
-            &context.commit5a_vec.as_ref().unwrap(),
+            &context
+                .commit5a_vec
+                .as_ref()
+                .ok_or(TssError::UnknownError {
+                    msg: ("gg18_sign_client_round6").to_string(),
+                    file: file!().to_string(),
+                    line: line!(),
+                })?
+                .clone(),
             &phase_5a_elgamal_vec,
             &phase_5a_dlog_vec,
-            &context.phase_5a_decom.as_ref().unwrap().V_i,
-            &context.r.as_ref().unwrap(),
+            &context
```

Comment:
Because we use a universal context containing all the messages between each round, lots of fields are None at some point.  Try to split the context into the immutable part and the mutable part. For the mutable part, it can be simply represented by a string,  at the end of each round,  we encode the message into a string, and then parse it at the beginning of the next round.  so this would reduce most of the `as_ref().unwrap()`